django/db/models/query_utils.py

```diff
@@ -193,19 +193,43 @@ def _check_parent_chain(self, instance):
         return None
 
 
+class classorinstancemethod:
+    """
+    A class used in RegisterLookupMixin which returns the necessary partial
+    functions depending on whether it was called by a Field instance or a Field
+    class
+    """
+
+    def __init__(self, class_method, instance_method):
+        self.class_method = class_method
+        self.instance_method = instance_method
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return functools.partial(self.class_method, owner)
+        return functools.partial(self.instance_method, instance)
+
+
 class RegisterLookupMixin:
-    @classmethod
-    def _get_lookup(cls, lookup_name):
-        return cls.get_lookups().get(lookup_name, None)
+    def _get_lookup(self, lookup_name):
+        return self.get_lookups().get(lookup_name, None)
 
-    @classmethod
     @functools.lru_cache(maxsize=None)
-    def get_lookups(cls):
+    def get_class_lookups(cls):
         class_lookups = [
             parent.__dict__.get("class_lookups", {}) for parent in inspect.getmro(cls)
         ]
         return cls.merge_dicts(class_lookups)
 
+    def get_instance_lookups(self):
+        class_lookups = self.get_class_lookups()
+        instance_lookups = self.__dict__.get("instance_lookups", {})
+        lookups = {**class_lookups, **instance_lookups}
+        return lookups
```

Comment:
Since the existence of instance lookups will be rare I suggest we avoid an unncessary dict creation when `instance_lookups` is missing

```suggestion
        if not (instance_lookups := self.__dict__.get("instance_lookups"):
            return class_lookups
        return {**class_lookups, **instance_lookups}
```